p2p/src/authenticated/actors/tracker/actor.rs

```diff
@@ -437,11 +478,17 @@ impl<E: Spawner + Rng + Clock + GClock + Metrics, C: Scheme> Actor<E, C> {
                     mut peer,
                 } => {
                     // Kill if peer is not authorized
-                    if !self.peers.contains_key(&public_key) {
+                    if !self.valid_peer(&public_key) {
                         peer.kill().await;
                         continue;
                     }
 
+                    // Peer is now active
```

Comment:
Construct is called every ~60s. I believe this check will panic on any long-lived connection.

Reply:
Ah okay

Reply:
Here is the related: https://github.com/commonwarexyz/monorepo/blob/103caa03ef469272b29f9f89a35bf27bd2d8f8a9/p2p/src/authenticated/actors/peer/actor.rs#L126-L127

Reply:
If you want to make that clearer in this code, feel free.